package.json

```diff
@@ -83,15 +80,18 @@
     "react-popper": "^2.2.4",
     "react-redux": "^7.2.1",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "^3.4.3",
+    "react-scripts": "^4.0.3",
     "react-text-mask": "^5.4.3",
     "redux": "^4.0.5",
     "redux-devtools-extension": "^2.13.8",
     "redux-thunk": "^2.1.0",
     "redux-undo": "^1.0.1",
     "styled-components": "^5.2.3",
-    "typescript": "^3.9.7",
-    "wait-on": "^5.2.1"
+    "typescript": "^4.2.4",
```

Comment:
Wcześniej mieliśmy wersję 3.9.7
O tym jakie zmiany zostały wprowadzone można poczytać w [dokumentacji](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html) zaczynając od tej wersji 4.0.0 i idąc w górę 
Niżej podaję zmiany które według mnie są najfajniejsze:
## Etykiety na elementach krotek
Pozwala nam dodawać etykiety do poszczególnych typów poszczególnych elementów krotek, co o wiele zwiększa czytelność naszego kodu w tych miejscach w których my ich używamy. 
![image](https://user-images.githubusercontent.com/21079319/116312943-645e1400-a7ad-11eb-8488-b1a6592b9344.png)

## Skrócone operatory przypisywania (Short-Circuiting Assignment Operators)
```ts
// before 
a = a ?? 15;
// after
a ??= 15;
```

## Template Literal Types
Teraz można definiować typ jako template-string. 
Czy to jest przydatne? 
Raczej tak. 
Na pewno bardzo fajnie wygląda :) 
![image](https://user-images.githubusercontent.com/21079319/116313813-8c01ac00-a7ae-11eb-82ac-6003a4c21675.png)